### PR TITLE
[Explore] highlighting run query when chart is stale on explore view

### PR DESCRIPTION
**Pull request by gabe-lyons**
This PR does two things:

1) makes the Run Query grey by default- we don't want to highlight this button unless it would actually provide a new result.

2) highlights the Run Query button when a change to the query powering the chart is made. Currently I determine a change is affecting the query if `renderTrigger` is falsey in the component's config. 

open questions:
1. I think whether we should trigger an instant refresh needs to be fleshed out a little bit more- for example moving a column from series to breakdowns doesn't affect the query but also doesn't trigger a rerender. Maybe we want to ditch the component level renderTrigger annotation and just look at the diff in the query (ignoring timestamp or something?) cc @graceguo-supercat @mistercrunch 
2. are there instances the chart should be updated that I'm not covering?

Before query change button is blank:
image

After query change button pops:
image

reviewers:
@graceguo-supercat 
@michellethomas 
@mistercrunch

### CODE REVIEW

**Comment by GabeLoins** [comment]
For future work I want to make it more obvious that the chart should be refreshed by overlaying it with a refresh button.

**Comment by michellethomas** [comment]
I like this idea, this generally looks good but Grace is more familiar with this part of the code and may know more if there are any edge cases we aren't thinking of.

File: superset/assets/javascripts/explore/components/ExploreViewContainer.jsx [hunk 1]
```diff
@@ -121,14 +126,22 @@ class ExploreViewContainer extends React.Component {
...
-  instantControlChanged(prevControls, currentControls) {
+  hasDisplayControlChanged(prevControls, currentControls) {
     return Object.keys(currentControls).some(key => (
       currentControls[key].renderTrigger &&
       typeof prevControls[key] !== 'undefined' &&
       !areObjectsEqual(currentControls[key].value, prevControls[key].value)
     ));
   }
 
+  hasQueryControlChanged(prevControls, currentControls) {
```
Comment (graceguo-supercat) [inline superset/assets/javascripts/explore/components/ExploreViewContainer.jsx]: hasDisplayControlChange and hasQueryControlChange has similar logic, can you consolidate this 2 functions? Notice both 2 functions called areObjectsEqual, which is a function that stringify 2 JS object and then find diff. Similar to JSON.parse, this function call is not cheap, so we need to use it with caution.

Also in componentWillReceiveProps we have a few logic block, check if viz_type changed, if data source changed, if control props changed, etc. The call to action creator, this.props.actions.ABCD, is not exit or return. Every time there is a props change, *all* of these logic will be checked. So if we do not consolidate 2 functions, the expensive compare function will be called twice every time there is a props change, no matter what kind of prop change.

**Comment by GabeLoins** [comment]
@graceguo-supercat addressed! Ready to merge.

@mistercrunch yeah, thats coming in PR #2

**Pull request merged**